06_puppy/marksomething/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import "sync"
+
+// Puppy represents a small dog
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+// PuppyStorer stores an object in a key-value store
+type PuppyStorer interface {
+	CreatePuppy(p Puppy)
+	ReadPuppy(k uint32) Puppy
+	UpdatePuppy(k uint32, p Puppy)
+	DeletePuppy(k uint32)
+}
+
+// MapStore is a storage mechanism for Puppy
+type MapStore map[uint32]Puppy
+
+// CreatePuppy adds a Puppy to storage unless a Puppy of the same ID is stored in which case nothing is done
+func (m MapStore) CreatePuppy(p Puppy) {
+	if _, found := m[p.ID]; found {
```

Comment:
`found` should be `ok` ([comma ok idiom](https://golang.org/doc/effective_go.html#maps))